lib/ohai/plugins/linux/network.rb

```diff
@@ -109,7 +109,15 @@ def check_routing_table(family, iface)
 
         # a sanity check, especially for Linux-VServer, OpenVZ and LXC:
         # don't report the route entry if the src address isn't set on the node
-        next if route_entry[:src] and not iface[route_int][:addresses].has_key? route_entry[:src]
+        # unless the interface has no addresses of this type at all
+        if route_entry[:src]
+          addr = iface[route_int][:addresses]
+          unless addr.nil? or addr.has_key? route_entry[:src] or
+                 not addr.values.any? { |a| a['family'] == family[:name] }
```

Comment:
Sorry - I was following the convention of the existing code. Will rework my diffs.